geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegion.java

```diff
@@ -557,6 +557,32 @@ public boolean virtualPut(EntryEventImpl event, boolean ifNew, boolean ifOld,
     }
   }
 
+  @Override
+  void cmnClearRegion(RegionEventImpl regionEvent, boolean cacheWrite, boolean useRVV) {
+    if (!getBucketAdvisor().isPrimary()) {
+      logger.info("Not primary bucket when doing clear, do nothing");
+      return;
+    }
+
+    boolean enableRVV = useRVV && getConcurrencyChecksEnabled();
+    RegionVersionVector rvv = null;
+    if (enableRVV) {
+      rvv = getVersionVector();
+    }
+
+    // get rvvLock
+    Set<InternalDistributedMember> participants =
+        getCacheDistributionAdvisor().adviseInvalidateRegion();
+    try {
+      obtainWriteLocksForClear(regionEvent, participants);
+      clearRegionLocally(regionEvent, cacheWrite, rvv);
+      distributeClearOperation(regionEvent, rvv, participants);
+
+      // TODO: call reindexUserDataRegion if there're lucene indexes
```

Comment:
I see you made reindexUserDataRegion public. Is this something you need to call here?

Reply:
Yes, that will be called in future as a todo work.

Reply:
But I changed it back for the time being